pycloudlib/ibm_softlayer/cloud.py

```diff
@@ -0,0 +1,492 @@
+# This file is part of pycloudlib. See LICENSE file for license information.
+"""IBM Cloud type."""
+
+from typing import List, Literal, Optional
+
+import SoftLayer  # type: ignore
+
+from pycloudlib.cloud import BaseCloud
+from pycloudlib.config import ConfigFile
+from pycloudlib.ibm_softlayer.errors import IBMSoftlayerException
+from pycloudlib.ibm_softlayer.instance import IBMSoftlayerInstance
+from pycloudlib.instance import BaseInstance
+
+
+class IBMSoftlayer(BaseCloud):
+    """IBM Softlayer Class."""
+
+    _type = "ibmsoftlayer"
+
+    VALID_REGIONS = [
```

Comment:
Instead of hardcoding a set of valid regions, couldn't we try-except on the sdk calls that use the region and fail? With a hardcoded list we will need to update it every time it changes.